src/DotNetWorker/Hosting/WorkerMiddlewareWorkerApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,104 @@
+        /// </list>
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseDefaultWorkerMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            return builder.UseOutputBindingsMiddleware()
+                .UseFunctionExecutionMiddleware();
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="FunctionExecutionMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseFunctionExecutionMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            builder.Services.AddSingleton<FunctionExecutionMiddleware>();
+
+            builder.Use(next =>
+            {
+                return context =>
+                {
+                    var middleware = context.InstanceServices.GetRequiredService<FunctionExecutionMiddleware>();
+
+                    return middleware.Invoke(context);
+                };
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="OutputBindingsMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        internal static IFunctionsWorkerApplicationBuilder UseOutputBindingsMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            builder.Services.AddSingleton<OutputBindingsMiddleware>();
+
+            builder.Use(next =>
+            {
+                return context =>
+                {
+                    var middleware = context.InstanceServices.GetRequiredService<OutputBindingsMiddleware>();
+
+                    return middleware.Invoke(context, next);
+                };
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="OutputBindingsMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        internal static IFunctionsWorkerApplicationBuilder UseMiddleware<T>(this IFunctionsWorkerApplicationBuilder builder)
```

Comment:
New extension method to make it easier to register custom middleware

Reply:
If this is internal, should IFunctionsWorkerMiddleware also be internal?

Reply:
The intention was for this to be public....